### PR TITLE
Fixes #18999: when we install a plugin, if the dependency (package manager) is not met, it still tries to install it and fails

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/18999

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

File: relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py [hunk 1]
```diff
@@ -307,7 +318,36 @@ def install_dependencies(metadata):
             logger.warning("The binary " + executable + " was not found on the system, you must install it before installing " + metadata['name'])
             return False
       else:
```
Comment (Fdall) [inline relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py:320]: You should add a case in the same way that the "binary" type was defined. I am pretty sure we want to have typed dependencies and so, an "apt" and a "rpm" type of dependencies..

**Review by Fdall — CHANGES_REQUESTED** [review]

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**